src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/UseAsyncInAsyncTests.cs

```diff
@@ -0,0 +1,1416 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information. 
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Testing;
+using Test.Utilities;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.UseAsyncInAsync,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.UseAsyncInAsync,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
+{
+    public class UseAsyncInAsyncTests
+    {
+
+        [Fact]
+        public async Task TaskWaitInTaskReturningMethodGeneratesWarning()
+        {
+            var testCS = @"
+using System.Threading.Tasks;
+
+class Test {
+    Task T() {
+        Task t = null;
+        [|t.Wait()|];
+        return Task.FromResult(1);
+    }
+}
+";
+            await VerifyCS.VerifyAnalyzerAsync(testCS);
+
+            var testVB = @"
+Imports System.Threading.Tasks
+
+Module Program
+    Sub Main()
+        Test()
+    End Sub
+    Function Test() As Task
+        Dim t As Task = Nothing
+        [|t.Wait()|]
+        Return Task.FromResult(1)
+    End Function
+End Module
+";
+            await VerifyVB.VerifyAnalyzerAsync(testVB);
+        }
+
+        [Fact]
+        public async Task TaskWaitInValueTaskReturningMethodGeneratesWarning()
+        {
+            var testCS = @"
+using System.Threading.Tasks;
+
+class Test {
+    ValueTask T() {
+        Task t = null;
+        [|t.Wait()|];
+        return default;
+    }
+}
+";
+            await VerifyCS.VerifyAnalyzerAsync(testCS);
+
+            var testVB = @"
+Imports System.Threading.Tasks
+
+Module Program
+    Sub Main()
+        Test()
+    End Sub
+    Function Test() As ValueTask
+        Dim t As Task = Nothing
+        [|t.Wait()|]
+        Return CType(Nothing, ValueTask)
+    End Function
+End Module
+";
+            await VerifyVB.VerifyAnalyzerAsync(testVB);
+        }
+
+        [Fact]
+        public async Task TaskWait_InIAsyncEnumerableAsyncMethod_ShouldReportWarning() // Feature 'async streams' is not available in C# 7.3. Please use language version 8.0 or greater.
```

Comment:
The analyzer reports the diagnostic correctly in this unit test, but I'm also getting another warning for `async streams` not being available in C# 7.3

Reply:
Do a Search in File for "LanguageVersion" and look at the results under the tests/* directory. There are many tests that set their language version to something more recent than 7.3